tests/test_sortedLinkedList.py

```diff
@@ -0,0 +1,220 @@
+
+    # We can add multiple items
+    assert sortedLinkedList.add(42)
+    assert sortedLinkedList.add(43)
+
+    assert sortedLinkedList.count() == 2
+
+    # Head and tail point to two different nodes
+    assert sortedLinkedList.getHead() == 43
+    assert sortedLinkedList.getTail() == 42
+
+    # We can get the prev node value for an item
+    assert sortedLinkedList.getPrev(43) == 42
+
+    # We can get the next node value for an item
+    assert sortedLinkedList.getNext(42) == 43
+
+    # Validate order
+    doOrderValidation(sortedLinkedList)
+
+    # We can remove a specific item by value and the other remains
+    assert sortedLinkedList.remove(42)
+    assert sortedLinkedList.count() == 1
+
+    assert sortedLinkedList.getTail() == 43
+    assert sortedLinkedList.getHead() == 43
+
+    # Validate order
+    doOrderValidation(sortedLinkedList)
+
+def test_sortedLinkedListAddMultipleItems(sortedLinkedListContractsFixture):
+
+    sortedLinkedList = sortedLinkedListContractsFixture.contracts['sortedLinkedList']
+
+    # Let's add many items and confirm the behavior is as expected
+    assert sortedLinkedList.add(42)
+    assert sortedLinkedList.add(41)
+    assert sortedLinkedList.add(43)
+
+    assert sortedLinkedList.count() == 3
+
+    assert sortedLinkedList.getHead() == 43
+    assert sortedLinkedList.getTail() == 41
+
+    assert sortedLinkedList.getNext(42) == 43
+    assert sortedLinkedList.getPrev(42) == 41
+
+    assert sortedLinkedList.getNext(41) == 42
+    assert sortedLinkedList.getPrev(43) == 42
+
+    assert sortedLinkedList.hasNext(43) == 0
+    assert sortedLinkedList.hasPrev(41) == 0
+
+    # Add to the ends and inbetween
+    assert sortedLinkedList.add(45)
+    assert sortedLinkedList.add(35)
+    assert sortedLinkedList.add(40)
+
+    # Validate order
+    doOrderValidation(sortedLinkedList)
+
+    assert sortedLinkedList.getPrev(45) == 43
+    assert sortedLinkedList.getNext(35) == 40
+    assert sortedLinkedList.getNext(40) == 41
+    assert sortedLinkedList.getPrev(41) == 40
+
+    assert sortedLinkedList.getTail() == 35
+    assert sortedLinkedList.getHead() == 45
+
+    assert sortedLinkedList.count() == 6
+
+    # Validate order
+    doOrderValidation(sortedLinkedList)
+
+    # Remove an inbetween
+    assert sortedLinkedList.remove(42)
+    assert sortedLinkedList.getPrev(43) == 41
+    assert sortedLinkedList.getNext(41) == 43
+
+    # Validate order
+    doOrderValidation(sortedLinkedList)
+
+    # Remove ends
+    assert sortedLinkedList.remove(35)
+    assert sortedLinkedList.getTail() == 40
+    assert sortedLinkedList.hasPrev(40) == 0
+
+    # Validate order
+    doOrderValidation(sortedLinkedList)
+
+    assert sortedLinkedList.remove(45)
+    assert sortedLinkedList.getHead() == 43
+    assert sortedLinkedList.hasNext(43) == 0
+
+    # Validate order
+    doOrderValidation(sortedLinkedList)
+
+def test_sortedLinkedListHints(sortedLinkedListContractsFixture):
```

Comment:
Recommend at least one test with multiple hints, ideally one where some hint from the middle is the one picked.